pallene/checker.lua

```diff
@@ -587,8 +637,11 @@ function FunChecker:check_var(var)
             var._type = self.p.module.functions[cname.id].typ
         elseif cname._tag == "checker.Name.Builtin" then
             var._type = builtins[cname.name].typ
-               else
-            error("impossible")
+        elseif cname._tag == "checker.Name.Module" then
+            -- The type for a module is a table.
+            var._type = builtin_modules[cname.name]
+        else
+            error("Unexpected condition: cname._tag == " .. cname._tag)
```

Comment:
Let's leave this as `error("impossible")` for now. I think it will be cleaner if we make this kind of change in a separate PR and if we change all the `error("imposible")`s at once.

Reply:
Okay. I'll revert it.